compiler/src/iree/compiler/InputConversion/StableHLO/Preprocessing/StableHLOToStableHLO.cpp

```diff
@@ -1540,6 +1540,40 @@ struct CustomCallIsTopK final
   }
 };
 
+// Recursive helper function that identifies an Iota followed by a set of
+// broadcasts where the last dimension of the iota is preserved throughout.
+bool isIotaOrIotaBroadcast(PatternRewriter &rewriter, Value input) {
+  auto iotaOp =
+      dyn_cast_or_null<mlir::stablehlo::IotaOp>(input.getDefiningOp());
+  if (iotaOp) {
+    int64_t iotaDim = iotaOp.getIotaDimension();
+    auto iotaLastDim = cast<ShapedType>(iotaOp.getType()).getRank() - 1;
+    if (iotaDim == iotaLastDim) {
+      return true;
+    } else {
```

Comment:
You can get rid of the `else` part as we returned in the `if` case.

Reply:
Good point, thanks!